src/components/molecules/TableHeader/TableHeader.tsx

```diff
@@ -138,60 +149,107 @@ const TableHeader: React.FC<TableHeaderProps> = ({
     };
   }, [leaveSeat]);
 
+  const updateTableTitle = useCallback(
+    async (newTableTitle) => {
+      if (!user) return;
+
+      const venueTables = tables.map((table) => {
+        if (table.reference === tableOfUser?.reference) {
+          return { ...table, title: newTableTitle };
+        }
+        return table;
+      });
+
+      // Ideally we want to do this only when config.tables doesn't exist. Otherwise we want to update only a single table.
+      await updateVenue_v2(
+        { name: venueName, tables: venueTables },
+        user
+      ).catch(() => {
+        // @debt Replace generic error when proper error handling is added
+        setError(GENERIC_ERROR);
+
+        const msg = "[updateVenue_v2] updating tables in venue.config.tables";
+        const context = {
+          location: "molecules::TableHeader::updateTableTitle",
+        };
+
+        console.warn(msg, context);
+        Bugsnag.notify(msg, (event) => {
+          event.severity = "warning";
+          event.addMetadata("context", context);
+        });
```

Comment:
This should probably be wrapped into the `updateVenue_v2` at the `api/*` layer like we do in most other places. (and then the error re-thrown so that the error state can react to it out here at the component level)

I believe we may also include other context at times too, such as the params being sent (there should be an example of this somewhere in the code, as I saw the Bugsnag side of it today, and the params made it much simpler to identify the root cause of the bug)

Also `molecules::TableHeader::updateTableTitle` is our older style. `molecules/TableHeader::updateTableTitle` is our latest style (pathname using `/`, function name/etc using `::`)

Reply:
Sounds good, I will move this to the `api` function.

Reply:
Moved in: https://github.com/sparkletown/sparkle/pull/1377/commits/de659957079dd74136f26d29e9b952be12add510

Reply:
Looks good except that you're not re-`throw`ing the error after the `Bugsnag.notify`, so as far as any downstream code is concerned, there is no error to `.catch` (if you think otherwise/can prove that it's not needed, would love to see that evidence, as this feels hacky AF)

Reply:
Just tested it, needs to re-`throw`:
![image](https://user-images.githubusercontent.com/753891/118237459-af309900-b4da-11eb-8d0a-a1fae5a409d7.png)

Reply:
The error is now thrown: https://github.com/sparkletown/sparkle/pull/1377/commits/4b7532e6f02068126c212a6f0dab1714aaaad55a